nntrainer/layers/bn_layer.cpp

```diff
@@ -228,8 +204,19 @@ void BatchNormalizationLayer::calcDerivative(RunLayerContext &context) {
     Tensor &dgamma = context.getWeightGrad(wt_idx[BNParams::gamma]);
     t_full.multiply_i(invstd);
     t_full.sum(axes_to_reduce, dgamma);
+
+    /**
+     * This implementation depends on the pre-calculated dbeta calculated.
+     */
+    Tensor &dbeta = context.getWeightGrad(wt_idx[BNParams::beta]);
+    t_reduced = dbeta.divide(divider);
```

Comment:
this new memory creation is solved universally for all layers at once in the new PR I am working on.
I verified locally that reusing this memory has no issue.